### PR TITLE
[FIX] teams.removeMembers mobile usage

### PR DESCRIPTION
**Pull request by reinaldonetof**
## Proposed changes

This PR https://github.com/RocketChat/Rocket.Chat/pull/23701 changed how to receive some params and this changed the behavior when removing members from a team. And if the user is the last owner in the team and he tries to leave, the error returned from the server is `invalid-params`, different than the expected, that was `Last owner cannot be removed`.

## How to test or reproduce

To test both servers you can use the `open.rocket.chat as 4.2.0` and `mobile.rocket.chat as 4.1`

Remove members from a team:
- Create a team, then add a member to the team and remove them from the team

Leave from a team that you are the owner:
- Create a team, add a channel to the team, then click on the title header, then click on `Leave`, the alert needs to be `Last owner cannot be removed`

## Screenshots
<table>
  <thead>
    <tr>
      <th>Before</th>
      <th>After</th>
      </thead>
    </tr>
    <tbody>
      <tr>
        <td>
        image
        </td>
        <td>
        image
        </td>
      </tr>
    </tbody>
</table >

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/lib/rocketchat.js [hunk 1]
```diff
@@ -836,16 +836,20 @@ const RocketChat = {
...
-	removeTeamMember({ teamId, teamName, userId, rooms }) {
+	removeTeamMember({ teamId, userId, rooms }) {
+		const serverVersion = reduxStore.getState().server.version;
 		// RC 3.13.0
 		return this.post('teams.removeMember', {
 			teamId,
-			teamName,
 			userId,
-			rooms
+			...((compareServerVersion(serverVersion, '4.2.0', methods.lowerThan) || rooms?.length) && { rooms })
```
Comment (diegolmello) [inline app/lib/rocketchat.js:852]: Why 4.2.0 here and 4.2 above?

File: app/lib/rocketchat.js [hunk 2]
```diff
@@ -836,16 +836,20 @@ const RocketChat = {
 		return this.post('teams.removeRoom', { roomId, teamId });
 	},
 	leaveTeam({ teamName, rooms }) {
+		const serverVersion = reduxStore.getState().server.version;
 		// RC 3.13.0
-		return this.post('teams.leave', { teamName, rooms });
+		return this.post('teams.leave', {
+			teamName,
+			...((compareServerVersion(serverVersion, '4.2', methods.lowerThan) || rooms?.length) && { rooms })
```
Comment (diegolmello) [inline app/lib/rocketchat.js:843]: I'm sorry, but I didn't get this logic.
- If the server is 4.1, it's going to use `rooms`
- If `rooms` are send, it's going to use `rooms`
So the only way not to use rooms, is server 4.2+ and not selecting any room on leave, but then why testing the server?
Isn't testing `rooms?.length` enough?

Comment (reinaldonetof) [reply app/lib/rocketchat.js]: For servers lower than 4.2.0 we must send rooms as param, at least an empty array. However the server 4.2.0 expects a minimum of 1 item inside the rooms' array, but he didn't accept if the array is empty, so if the array is empty we can't pass rooms as param.

**Review by AlexAlexandre — APPROVED** [review]
LGTM

**Review by AlexAlexandre — APPROVED** [review]
LGTM

**Pull request merged**